Sources/StreamChat/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -40,47 +41,94 @@ final class CurrentUserController_Tests: StressTestCase {
     }
     
     // MARK: Controller
-
-    func test_initialState_whenLocalDataIsFetched() throws {
-        let unreadCount = UnreadCount(channels: 10, messages: 212)
-        let userPayload: CurrentUserPayload<NoExtraData> = .dummy(userId: .unique, role: .user, unreadCount: unreadCount)
-
-        // Save user to the db
-        try client.databaseContainer.writeSynchronously {
-            try $0.saveCurrentUser(payload: userPayload)
-        }
+    
+    // MARK: - Synchronize tests
+    
+    func test_synchronize_localDataIsAvailable() {
+        let expectedId = UserId.unique
+        let expectedUnreadCount = UnreadCount(channels: 10, messages: 212)
+        
+        env.currentUserObserverItem = .init(id: expectedId, unreadCount: expectedUnreadCount)
+        
+        controller.synchronize()
         
         // Assert client is assigned correctly
         XCTAssertTrue(controller.client === client)
         
         // Assert user is correct
-        XCTAssertEqual(controller.currentUser?.id, userPayload.id)
+        XCTAssertEqual(controller.currentUser?.id, expectedId)
         
         // Assert unread-count is correct
-        XCTAssertEqual(controller.unreadCount, unreadCount)
+        XCTAssertEqual(controller.unreadCount, expectedUnreadCount)
     }
     
-    func test_initialState_whenLocalDataFetchFailed() throws {
-        let unreadCount = UnreadCount(channels: 10, messages: 212)
-        let userPayload: CurrentUserPayload<NoExtraData> = .dummy(userId: .unique, role: .user, unreadCount: unreadCount)
+    func test_synchronize_changesControllerState() {
+        // Check if controller has initialized state initially.
+        XCTAssertEqual(controller.state, .initialized)
         
-        // Save user to the db
-        try client.databaseContainer.writeSynchronously {
-            try $0.saveCurrentUser(payload: userPayload)
-        }
+        // Simulate current user
+        env.currentUserObserverItem = .init(id: .unique)
+        
+        // Simulate `synchronize` call.
+        controller.synchronize()
+        
+        // Simulate successful network call.
+        env.currentUserUpdater.updateDevices_completion?(nil)
+        
+        // Check if state changed after successful network call.
+        XCTAssertEqual(controller.state, .remoteDataFetched)
+        XCTAssertNotNil(env.currentUserUpdater.updateDevices_currentUserId)
+    }
+    
+    func test_synchronize_whenNoCurrentUser_doesNotMakeRemoteCall() {
+        // Check if controller has initialized state initially.
+        XCTAssertEqual(controller.state, .initialized)
+        
+        // Simulate no current user
+        env.currentUserObserverItem = nil
+        
+        // Simulate `synchronize` call.
+        controller.synchronize()
+        
+        // Check if state changed after successful network call.
+        XCTAssertEqual(controller.state, .localDataFetched)
```

Comment:
This comment doesn't seem correct. Also, you should check that the synchronize call returns an error when there's no current user.